Source/ResponseSerialization.swift

```diff
@@ -575,11 +583,11 @@ public final class DecodableResponseSerializer<T: Decodable>: ResponseSerializer
                 throw AFError.responseSerializationFailed(reason: .inputDataNilOrZeroLength)
             }
 
-            guard let emptyValue = Empty.value as? T else {
+            guard let emptyResponseType = T.self as? EmptyResponse.Type else {
```

Comment:
It seems like there should be a more elegant want to handle this. At the least, perhaps combining the type cast and the value cast into the guard?

Reply:
I updated the guard in 9a7bb81 just now. It'll take me a little time to get to writing the tests. Thanks for seeing the value in this.